### ISSUE TITLE
Ray Wait is waiting for all tasks to complete

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#152]
## RL Experiments

I'm doing RL sampling right now, and I'm timing the tasks at a finer grain. I'm also using a ray version off commit 6cd02d71f8e
```
    while num_samples < max_samples:
        for i in range(num_workers - len(remaining)): 
            remaining.append(ray_rollout.remote(param_id, max_path_length))
        done, remaining = ray.wait(remaining)
        result, timestamps = ray.get(done[0])
        num_samples += len(result['rewards'])
```

Here is the rollout code: 
```
@ray.remote
def ray_rollout(policy_params, max_path_length):
    """returns rollout dictionary, id, (start, end)"""
    start_time = str(datetime.now())
    ...
    return rollout(policy_params), (start_time, str(datetime.now()))
```
However, instead of getting fine-grain tasks, the time plots look like this:

![screenshot 2016-12-23 14 20 12](https://cloud.githubusercontent.com/assets/4529381/21463259/61fd6a8c-c91b-11e6-8227-20e9a031e5c3.png)

**Comment by robertnishihara** [comment]
Thanks for catching this! This should be fixed by #153.

**Comment by robertnishihara** [comment]
Closing, but please reopen if the bug persists.
